dulwich/sparse_patterns.py

```diff
@@ -0,0 +1,462 @@
+# sparse_patterns.py -- Sparse checkout pattern handling.
```

Comment:
Please add tests for each of the functions/classes in this file.

Reply:
yep, this one is nice and simple to test as the file is just a readable list of patterns, will do tests next